src/ClpSolverInterface.jl

```diff
@@ -237,3 +237,195 @@ end
+    ubref = CR{MOI.ScalarVariablewiseFunction, MOI.GreaterThan{Float64}}(nc)
+    push!(m.vartolbref, lbref)
+    push!(m.vartoubref, ubref)
+    m.lbreftovar[lbref] = nv
+    m.ubreftovar[ubref] = nv
+    return vf
+end
+
+function MOI.addvariables!(m::ClpMOIModel, n::Integer)
+    return [MOI.addvariable!(m) for i in 1:n]
+end
+
+function MOI.delete!(m::ClpMOIModel, vr::VR)
+    which = Vector{Int32}()
+    push!(which, col(m,vr))
+    delete_columns(m.inner, which)
+end
+
+
+# Objective
+function MOI.setobjective!(m::ClpMOIModel, sense::MOI.OptimizationSense, f::MOI.ScalarAffineFunction{Float64})
+    obj_in = zeros(Float64, m.nvariables)
+    for i in 1:length(f.variables)
+        index = col(m,f.variables[i]) + 1
+        coeff = f.coefficients[i]
+        obj_in[index] = coeff
+    end
+    chg_obj_coefficients(m.inner, obj_in)
+    set_objective_offset(m.inner, f.constant)
+    if sense == MOI.MinSense
+        set_obj_sense(m.inner, 1.0)
+    elseif sense == MOI.MaxSense
+        set_obj_sense(m.inner, -1.0)
+    else
+        set_obj_sense(m.inner, 0.0)
+    end
+end
+
+# MOI.getattribute(m::ClpMOIModel, ::MOI.Sense)
+# MOI.getattribute(m::ClpMOIModel, ::MOI.ObjectiveFunction)
+# MOI.modifyobjective!(m::ClpMOIModel, change::MOI.AbstractFunctionModification)
+
+# Constraint
+function _addrangeref!(F,S,m::ClpMOIModel)
+    m.nconstraints += 1
+    m.nrows += 1
+    cf = CR{F,S}(m.nconstraints)
+    push!(m.rowtorangeref, cf)
+    m.rangereftorow[cf] = m.nrows
+    return cf
+end
+
+function _addrangeinsolver!(m::ClpMOIModel, f::MOI.ScalarAffineFunction{Float64}, lower::Float64, upper::Float64)
+    elements = Vector{Float64}()
+    columns = Vector{Int32}()
+    number_in_row = length(f.variables)
+    for i in 1:number_in_row
+        push!(columns, col(m,f.variables[i]))
+        push!(elements, f.coefficients[i])
+    end
+    add_row(m.inner, number_in_row, columns, elements, lower - f.constant, upper - f.constant)
+end
+
+function _addrange!(m::ClpMOIModel, f::MOI.ScalarAffineFunction{Float64}, lower::Float64, upper::Float64)
+    _addrangeinsolver!(m, f, lower, upper)
+    return _addrangeref!(MOI.ScalarAffineFunction{Float64},MOI.Interval{Float64},m)
+end
+
+function MOI.addconstraint!(m::ClpMOIModel, f::MOI.ScalarAffineFunction{Float64}, s::MOI.Interval{Float64})
+    return _addrange!(m, f, s.lower, s.upper)
+end
+
+function MOI.addconstraint!(m::ClpMOIModel, f::MOI.ScalarAffineFunction{Float64}, s::MOI.GreaterThan{Float64})
+    return _addrange!(m, f, s.lower, Inf)
+end
+
+function MOI.addconstraint!(m::ClpMOIModel, f::MOI.ScalarAffineFunction{Float64}, s::MOI.LessThan{Float64})
+    return _addrange!(m, f, -Inf, s.upper)
+end
+
+function replaceInf(x)
+    for i in 1:length(x)
+        if x[i] > 1e20
+            x[i] = Inf
+        elseif x[i] < -1e20
+            x[i] = -Inf
+        end
+    end
+    return x
+end
+
+function MOI.addconstraint!(m::ClpMOIModel, f::MOI.ScalarVariablewiseFunction, s::MOI.GreaterThan{<:Real})
```

Comment:
This implementation has a quadratic cost for setting all variable bounds. This will definitely be too slow. The interface should keep its own copy of the bounds and pass them to Clp when needed.